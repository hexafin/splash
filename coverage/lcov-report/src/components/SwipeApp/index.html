<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/SwipeApp</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> src/components/SwipeApp
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">56.61% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>137/242</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">25.66% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>29/113</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">37.7% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>23/61</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">55.7% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>132/237</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line medium'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="Balance.js"><a href="Balance.js.html">Balance.js</a></td>
	<td data-value="51.16" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 51%;"></div><div class="cover-empty" style="width:49%;"></div></div></td>
	<td data-value="51.16" class="pct medium">51.16%</td>
	<td data-value="43" class="abs medium">22/43</td>
	<td data-value="21.88" class="pct low">21.88%</td>
	<td data-value="32" class="abs low">7/32</td>
	<td data-value="30" class="pct low">30%</td>
	<td data-value="10" class="abs low">3/10</td>
	<td data-value="51.16" class="pct medium">51.16%</td>
	<td data-value="43" class="abs medium">22/43</td>
	</tr>

<tr>
	<td class="file medium" data-value="CurrencySwitch.js"><a href="CurrencySwitch.js.html">CurrencySwitch.js</a></td>
	<td data-value="59.46" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 59%;"></div><div class="cover-empty" style="width:41%;"></div></div></td>
	<td data-value="59.46" class="pct medium">59.46%</td>
	<td data-value="37" class="abs medium">22/37</td>
	<td data-value="6.25" class="pct low">6.25%</td>
	<td data-value="16" class="abs low">1/16</td>
	<td data-value="46.15" class="pct low">46.15%</td>
	<td data-value="13" class="abs low">6/13</td>
	<td data-value="58.33" class="pct medium">58.33%</td>
	<td data-value="36" class="abs medium">21/36</td>
	</tr>

<tr>
	<td class="file medium" data-value="Header.js"><a href="Header.js.html">Header.js</a></td>
	<td data-value="71.43" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 71%;"></div><div class="cover-empty" style="width:29%;"></div></div></td>
	<td data-value="71.43" class="pct medium">71.43%</td>
	<td data-value="7" class="abs medium">5/7</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="4" class="abs medium">2/4</td>
	<td data-value="71.43" class="pct medium">71.43%</td>
	<td data-value="7" class="abs medium">5/7</td>
	</tr>

<tr>
	<td class="file medium" data-value="ReturnToHome.js"><a href="ReturnToHome.js.html">ReturnToHome.js</a></td>
	<td data-value="78.57" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 78%;"></div><div class="cover-empty" style="width:22%;"></div></div></td>
	<td data-value="78.57" class="pct medium">78.57%</td>
	<td data-value="14" class="abs medium">11/14</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	<td data-value="78.57" class="pct medium">78.57%</td>
	<td data-value="14" class="abs medium">11/14</td>
	</tr>

<tr>
	<td class="file medium" data-value="SwipeApp.js"><a href="SwipeApp.js.html">SwipeApp.js</a></td>
	<td data-value="54.61" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 54%;"></div><div class="cover-empty" style="width:46%;"></div></div></td>
	<td data-value="54.61" class="pct medium">54.61%</td>
	<td data-value="141" class="abs medium">77/141</td>
	<td data-value="32.79" class="pct low">32.79%</td>
	<td data-value="61" class="abs low">20/61</td>
	<td data-value="36.67" class="pct low">36.67%</td>
	<td data-value="30" class="abs low">11/30</td>
	<td data-value="53.28" class="pct medium">53.28%</td>
	<td data-value="137" class="abs medium">73/137</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Apr 19 2019 16:14:10 GMT-0400 (Eastern Daylight Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
<script src="../../../block-navigation.js"></script>
</body>
</html>
