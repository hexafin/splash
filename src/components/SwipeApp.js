import React, { Component } from "react"
import {
	Dimensions,
	ScrollView,
	Image,
	View,
	Text,
	Animated,
	StyleSheet,
	TouchableWithoutFeedback
} from "react-native"
import firebase from "react-native-firebase"
import { colors } from "../lib/colors"
import { defaults, icons } from "../lib/styles"
import { isIphoneX } from "react-native-iphone-x-helper"
import { connect } from "react-redux";
import { bindActionCreators } from "redux"
import PropTypes from "prop-types"
import Account from "./Account"
import Receive from "./Receive"
import Home from "./Home"
import ReactNativeHapticFeedback from 'react-native-haptic-feedback';

const SCREEN_WIDTH = Dimensions.get("window").width
const SCREEN_HEIGHT = Dimensions.get("window").height

const xOffset = new Animated.Value(0)
const _onScroll = Animated.event(
	[{ nativeEvent: { contentOffset: { x: xOffset } } }],
	{
		useNativeDriver: true
	}
)

function Page(props: { children?: ReactElement<*> }) {
	return (
		<View style={{ flex: 1, width: SCREEN_WIDTH }}>{props.children}</View>
	)
}

function iconTransform(index: number) {
	return {
		transform: [
			{
				scale: xOffset.interpolate({
					inputRange: [
						(index - 1) * SCREEN_WIDTH,
						index * SCREEN_WIDTH,
						(index + 1) * SCREEN_WIDTH
					],
					outputRange: [0.75, 1, 0.75]
				})
			},
			{
				translateX: xOffset.interpolate({
					inputRange: [
						(index - 1) * SCREEN_WIDTH,
						index * SCREEN_WIDTH,
						(index + 1) * SCREEN_WIDTH
					],
					outputRange: [198, 0, -198]
				})
			}
		]
	}
}

function headerTransform() {
	return {
		transform: [
			{
				translateY: xOffset.interpolate({
					inputRange: [
						0,
						1 * SCREEN_WIDTH,
						2 * SCREEN_WIDTH
					],
					outputRange: [-20, 0, -20]
				})
			}
		]
	}
}

class SwipeApp extends Component {
	refScrollView = view => {
		if (view) {
			this.scrollView = view.getNode()
		}
	}

	constructor(props) {
		super(props)
		this.state = {
			activePage: this.props.activePage ? this.props.activePage : "Home",
			activeIndex: 1,
			isScrolling: false
		}
		this.pages = [
			{
				name: "Account",
				component: Account,
				image: icons.whiteSplash
			},
			{
				name: "Home",
				component: Home,
				image: null
			},
			{
				name: "Receive",
				component: Receive,
				image: icons.qrIcon
			}
		]
		this.pageIndices = {}
		for (let i = 0; i < this.pages.length; i++) {
			const page = this.pages[i]
			this.pageIndices[page.name] = i
		}
		this.goToPage = this.goToPage.bind(this)
		this.goToPageByIndex = this.goToPageByIndex.bind(this)
	}

	goToPage = (scrollView, pageName) => {
		try {
			const xOffset = SCREEN_WIDTH * this.pageIndices[pageName]
			scrollView.scrollTo({ x: xOffset, y: 0, animated: true })
			this.setState(prevState => {
				return {
					...prevState,
					activePage: pageName,
					activeIndex: this.pageIndices[pageName]
				}
			})
		}
		catch (error) {
			console.log(error)
		}
	}

	goToPageByIndex = (scrollView, index) => {
		try {
			if (index >= 0 && index < this.pages.length) {
				// console.log(index)
				const xOffset = SCREEN_WIDTH * index
				scrollView.scrollTo({ x: xOffset, y: 0, animated: true })
				this.setState(prevState => {
					return {
						...prevState,
						activePage: this.pages[index].name,
						activeIndex: index
					}
				})
			}
		}
		catch (error) {
			console.log(error)
		}
	}

	componentDidMount() {
		this.goToPageByIndex(this.scrollView, 1)
		// initialize swipe app to center page
		xOffset.setValue(SCREEN_WIDTH)
	}

	render() {

		const Pages = []
		const Icons = []
		for (let i = 0; i < this.pages.length; i++) {
			const page = this.pages[i]
			Pages.push(
				<Page key={"page-" + i}>
					{React.createElement(page.component, {
						...this.props,

					})}
				</Page>
			)
			Icons.push(
				<TouchableWithoutFeedback
					key={"page-icon-" + i}
					onPressIn={() => {
						ReactNativeHapticFeedback.trigger("impactLight", true)
						this.goToPageByIndex(this.scrollView, i)
					}}>
					<Animated.Image
						source={page.image}
						resizeMode="contain"
						style={[
							styles.icon,
							iconTransform(i),
							{
								height: (page.name == "Receive") ? 40 : 45,
								width: (page.name == "Receive") ? 40 : 45,
							}
						]}
					/>
				</TouchableWithoutFeedback>
			)
		}

		return (
			<View style={styles.container}>
				<Animated.Image
					source={require("../assets/images/headerWave.png")}
					resizeMode="contain"
					style={[headerTransform(), styles.headerImage]}/>
				<Animated.ScrollView
					horizontal
					pagingEnabled
					// ref={ref => (this.tabBar = ref._component)}
					ref={this.refScrollView}
					showsHorizontalScrollIndicator={false}
					showsVerticalScrollIndicator={false}
					automaticallyAdjustContentInsets={false}
					onScrollBeginDrag={event => {
						this.setState(prevState => {
							return {
								...prevState,
								isScrolling: true
							}
						})
					}}
					onMomentumScrollEnd={event => {
						try {
							const xOffset = event.nativeEvent.contentOffset.x
							const activePageName = this.pages[
								xOffset / SCREEN_WIDTH
							].name
							this.setState(prevState => {
								return {
									...prevState,
									activePage: activePageName,
									activeIndex: this.pageIndices[activePageName],
									isScrolling: false
								}
							})
						}
						catch (error) {
							console.log(error)
						}
					}}
					contentOffset={{
						x: SCREEN_WIDTH
					}}
					scrollEventThrottle={16}
					bounces={false}
					onScroll={_onScroll}
					style={{ flex: 1, flexDirection: "row" }}>
					{Pages}
				</Animated.ScrollView>
				
				{Icons}
			</View>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		backgroundColor: colors.white,
		flex: 1
	},
	icon: {
		position: "absolute",
		top: isIphoneX() ? 55 : 35,
		alignSelf: "center",
		overflow: "visible",
		marginBottom: 15,
		paddingLeft: 20,
		paddingRight: 20,
		shadowOffset: {
			width: 0,
			height: 5
		},
		shadowOpacity: 0.12,
		shadowRadius: 12
	},
	headerImage: {
		top: (isIphoneX()) ? -30 : -50,
		width: SCREEN_WIDTH,
		height: 240,
		position: "absolute"
	}
})


const mapStateToProps = state => {
	return {
		
	}
}

const mapDispatchToProps = dispatch => {
	return bindActionCreators(
		{
			
		},
		dispatch
	)
}

export default connect(mapStateToProps, mapDispatchToProps)(SwipeApp)
