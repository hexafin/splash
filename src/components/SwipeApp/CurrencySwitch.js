// @flow
import React, {Component} from 'react'
import {
  Platform, 
  StyleSheet, 
  TouchableWithoutFeedback,
  Text,
  Share,
  TextInput,
  Image,
  View,
  ScrollView,
  Keyboard,
  Animated,
  Dimensions
} from 'react-native'
import { isIphoneX } from "react-native-iphone-x-helper"
import ReactNativeHapticFeedback from 'react-native-haptic-feedback'
import { connect } from "react-redux"
import { cryptoNames, cryptoNameDict, cryptoColors, cryptoImages } from "../../lib/cryptos"
import { bindActionCreators } from "redux"
import { colors } from "../../lib/colors"
import Interactable from 'react-native-interactable';
const SCREEN_WIDTH = Dimensions.get("window").width
const SCREEN_HEIGHT = Dimensions.get("window").height

let currencies = []
let i = 0
let snapPoints = []
cryptoNames.forEach(crypto => {
  currencies.push({
    index: i,
    name: cryptoNameDict[crypto],
    code: crypto,
    image: cryptoImages[crypto]
  })
  snapPoints.push({
    x: -1 * (SCREEN_WIDTH/2-45) * i
  })
  i++
})

class CurrencySwitch extends Component {

  constructor(props) {
    super(props)
  }

  handleCurrencySwitch(currency) {
    console.log(currency)
  }

  render() {

    return (
      <Animated.View style={{
        ...styles.container,
        opacity: Animated.multiply(
          this.props.xOffset.interpolate({
            inputRange: [SCREEN_WIDTH*2/3, SCREEN_WIDTH, SCREEN_WIDTH*4/3],
            outputRange: [0, 1, 0]
          }),
          this.props.yOffsets.home.interpolate({
            inputRange: [0, 140, 160],
            outputRange: [1, 1, 0]
          })
        ),
        transform: [
          {translateY: Animated.add(
            this.props.yOffsets.home.interpolate({
              inputRange: [-81, -80, 0, 160, 161],
              outputRange: [50, 50, 0, -100, -300]
            }),
            this.props.xOffset.interpolate({
              inputRange: [0, SCREEN_WIDTH*2/3, SCREEN_WIDTH, SCREEN_WIDTH*4/3, SCREEN_WIDTH*2],
              outputRange: [-300, -50, 0, -50, -300]
            })
          )},
        ]
      }} pointerEvents="box-none">

        <Interactable.View
          ref={ref => this.interactable = ref}
          horizontalOnly={true}
          snapPoints={snapPoints}
          animatedValueX={this.props.switchXOffset}
          onSnap={event => {
            this.handleCurrencySwitch(currencies[event.nativeEvent.index].code)
          }}>

          <View style={styles.coinWrapper}>

            {currencies.map(currency => (
              <TouchableWithoutFeedback
                key={`currency${currency.code}`}
                onPress={() => {
                  this.interactable.snapTo({index: currency.index})
                }}>
                <Animated.View
                  style={{
                    ...styles.coin,
                    opacity: this.props.switchXOffset.interpolate({
                      inputRange: [
                        -1 * (SCREEN_WIDTH/2-45) * (currency.index+1), 
                        -1 * (SCREEN_WIDTH/2-45) * (currency.index),
                        -1 * (SCREEN_WIDTH/2-45) * (currency.index-1)],
                      outputRange: [0.5, 1, 0.5]
                    }),
                    transform: [
                      {scale: Animated.multiply(
                        this.props.yOffsets.home.interpolate({
                          inputRange: [-81, -80, 0, 60, 160, 161],
                          outputRange: [1.2, 1.2, 1, 0.95, 0.2, 0.2]
                        }),
                        this.props.switchXOffset.interpolate({
                          inputRange: [
                            -1 * (SCREEN_WIDTH/2-45) * (currency.index+1), 
                            -1 * (SCREEN_WIDTH/2-45) * (currency.index),
                            -1 * (SCREEN_WIDTH/2-45) * (currency.index-1)],
                          outputRange: [0.8, 1, 0.8]
                        })
                      )},
                    ]
                  }}>
                  <Image
                    source={currency.image}
                    resizeMode="contain"
                    style={styles.coinImage}/>
                  <Text style={styles.coinText}>{currency.code}</Text>
                </Animated.View>
              </TouchableWithoutFeedback>
            ))}

          </View>

        </Interactable.View>
      </Animated.View>
    );
  }
}

const styles = StyleSheet.create({
  coinWrapper: {
    flexDirection: "row",
    paddingLeft: (SCREEN_WIDTH - 100)/2,
    paddingRight: 50,
    // backgroundColor: colors.primary,
    width: "auto",
    justifyContent: "space-between"
  },
  coin: {
    width: 100,
    height: 120,
    marginRight: (SCREEN_WIDTH - 100)/2 - 95,
  },
  coinImage: {
    width: 100,
    height: 100,
  },
  coinText: {
    width: 100,
    textAlign: "center",
    fontSize: 22,
    fontWeight: "500",
    color: colors.white,
    marginTop: 5,
  },
  container: {
    position: "absolute",
    top: isIphoneX() ? 175 : 155,
  }
})

const mapStateToProps = state => {
  return {
    
  }
}

const mapDispatchToProps = dispatch => {
  return bindActionCreators(
    {
      
    },
    dispatch
  );
}

export default connect(mapStateToProps, mapDispatchToProps)(CurrencySwitch)
