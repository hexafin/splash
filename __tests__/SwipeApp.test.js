import "react-native";
import React from "react";
import SwipeApp, { onScroll } from "../src/components/SwipeApp/SwipeApp";
import { ReturnToHome } from "../src/components/SwipeApp/ReturnToHome";
import Header from "../src/components/SwipeApp/Header";
import { CurrencySwitch } from "../src/components/SwipeApp/CurrencySwitch";
import { Balance } from "../src/components/SwipeApp/Balance";

global.onScroll = onScroll;

const mockAnimated = {
	interpolate: () => {
		return 50;
	},
	__getValue: () => {
		return 50;
	},
	addListener: () => {}
};

it("Swipe App renders correctly", () => {
	const snap = shallow(
		<SwipeApp
			Load={() => {}}
			setNotifsRequested={() => {}}
			notificationPermissionInfo={() => {
				return Promise.resolve();
			}}
		/>
	);
	expect(snap).toMatchSnapshot();
});

it("Return To Home renders correctly", () => {
	const snap = shallow(
		<ReturnToHome
			xOffset={mockAnimated}
			yOffsets={{
				home: mockAnimated
			}}
		/>
	);
	expect(snap).toMatchSnapshot();
});

it("Header renders correctly", () => {
	const snap = shallow(
		<Header
			fill={mockAnimated}
			fillInput={mockAnimated}
			xOffset={mockAnimated}
			yOffsets={{
				home: mockAnimated
			}}
		/>
	);
	expect(snap).toMatchSnapshot();
});

it("Currency Switch renders correctly", () => {
	const snap = shallow(
		<CurrencySwitch
			activeCurrency={"USD"}
			activeCryptoCurrency={"BTC"}
			loading={false}
			xOffset={mockAnimated}
			yOffsets={{
				home: mockAnimated
			}}
			switchXOffset={mockAnimated}
		/>
	);
	expect(snap).toMatchSnapshot();
});

it("Balance renders correctly", () => {
	const snap = shallow(
		<Balance
			setActiveCurrency={() => {}}
			currency={"USD"}
			cryptoCurrency={"BTC"}
			balance={{
				BTC: 100,
				ETH: 0,
				GUSD: 0
			}}
			exchangeRates={{
				USD: {
					BTC: 100
				}
			}}
			isLoadingExchangeRates={false}
			isLoadingTransactions={false}
			isLoadingBalance={false}
			loadingBalanceCurrency={"ETH"}
			successLoadingBalance={true}
			xOffset={mockAnimated}
			yOffsets={{
				home: mockAnimated
			}}
		/>
	);
	expect(snap).toMatchSnapshot();
});
