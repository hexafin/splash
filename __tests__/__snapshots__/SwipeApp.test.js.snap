// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Balance renders correctly 1`] = `
<AnimatedComponent
  pointerEvents="box-none"
  style={
    Array [
      Object {
        "opacity": 50,
        "transform": Array [
          Object {
            "scale": 2550,
          },
          Object {
            "translateY": 2550,
          },
          Object {
            "translateX": 50,
          },
        ],
      },
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
        "position": "absolute",
        "top": 90,
        "width": 750,
      },
    ]
  }
>
  <AnimatedComponent
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "center",
        },
        Object {
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        },
      ]
    }
  >
    <Component
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#FFFFFF",
          "fontSize": 36,
          "fontWeight": "600",
        }
      }
    >
      0
    </Component>
    <Component
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
            "position": "absolute",
            "top": 0,
            "width": "100%",
          },
          Object {
            "opacity": 0,
          },
        ]
      }
    >
      <LoadingCircle
        restart={false}
        size={30}
      />
    </Component>
    <Connect(CurrencySwitcher)
      activeCurrencySize={24}
      crypto="BTC"
      fiat="USD"
      onPressIn={[Function]}
      onPressOut={[Function]}
      style={
        Object {
          "marginTop": -25,
        }
      }
      switcherBottom={32}
      switcherColor="white"
      textColor="rgba(255, 255, 255, 0.7)"
    />
  </AnimatedComponent>
</AnimatedComponent>
`;

exports[`Currency Switch renders correctly 1`] = `
<AnimatedComponent
  pointerEvents="box-none"
  style={
    Object {
      "opacity": 2500,
      "position": "absolute",
      "top": 175,
      "transform": Array [
        Object {
          "translateY": 100,
        },
      ],
    }
  }
>
  <Component
    animatedValueX={
      Object {
        "__getValue": [Function],
        "addListener": [Function],
        "interpolate": [Function],
      }
    }
    horizontalOnly={true}
    initialPosition={
      Object {
        "x": -0,
      }
    }
    snapPoints={
      Array [
        Object {
          "x": -0,
        },
        Object {
          "x": -330,
        },
        Object {
          "x": -660,
        },
      ]
    }
  >
    <Component
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingLeft": 325,
          "paddingRight": 50,
          "width": "auto",
        }
      }
    >
      <TouchableWithoutFeedback
        key="currencyBTC"
        onPress={[Function]}
      >
        <AnimatedComponent
          style={
            Object {
              "height": 120,
              "marginRight": 230,
              "opacity": 50,
              "transform": Array [
                Object {
                  "scale": 2500,
                },
              ],
              "width": 100,
            }
          }
        >
          <Component
            resizeMode="contain"
            source={
              Object {
                "testUri": "../../../src/assets/images/bitcoin-logo.png",
              }
            }
            style={
              Object {
                "height": 100,
                "width": 100,
              }
            }
          />
          <Component
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 22,
                "fontWeight": "500",
                "marginTop": 5,
                "textAlign": "center",
                "width": 100,
              }
            }
          >
            BTC
          </Component>
        </AnimatedComponent>
      </TouchableWithoutFeedback>
      <TouchableWithoutFeedback
        key="currencyETH"
        onPress={[Function]}
      >
        <AnimatedComponent
          style={
            Object {
              "height": 120,
              "marginRight": 230,
              "opacity": 50,
              "transform": Array [
                Object {
                  "scale": 2500,
                },
              ],
              "width": 100,
            }
          }
        >
          <Component
            resizeMode="contain"
            source={
              Object {
                "testUri": "../../../src/assets/images/ether-icon.png",
              }
            }
            style={
              Object {
                "height": 100,
                "width": 100,
              }
            }
          />
          <Component
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 22,
                "fontWeight": "500",
                "marginTop": 5,
                "textAlign": "center",
                "width": 100,
              }
            }
          >
            ETH
          </Component>
        </AnimatedComponent>
      </TouchableWithoutFeedback>
      <TouchableWithoutFeedback
        key="currencyGUSD"
        onPress={[Function]}
      >
        <AnimatedComponent
          style={
            Object {
              "height": 120,
              "marginRight": 230,
              "opacity": 50,
              "transform": Array [
                Object {
                  "scale": 2500,
                },
              ],
              "width": 100,
            }
          }
        >
          <Component
            resizeMode="contain"
            source={
              Object {
                "testUri": "../../../src/assets/images/gemini-icon.png",
              }
            }
            style={
              Object {
                "height": 100,
                "width": 100,
              }
            }
          />
          <Component
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 22,
                "fontWeight": "500",
                "marginTop": 5,
                "textAlign": "center",
                "width": 100,
              }
            }
          >
            GUSD
          </Component>
        </AnimatedComponent>
      </TouchableWithoutFeedback>
    </Component>
  </Component>
</AnimatedComponent>
`;

exports[`Header renders correctly 1`] = `
<[object Object]
  height={792}
  viewBox="0 0 375 396"
  width={750}
>
  <AnimatedComponent
    fill={50}
  >
    <Path
      d="M0,0 L375,0 L375,387.179245 C339.355325,393.704622 309.789941,396.485791 286.303847,395.522753 C225.328718,393.02249 169.084364,374.698552 110.07317,377.843259 C51.3417556,380.973056 14.6506989,385.197965 0,390.517986 L0,0 Z"
    />
  </AnimatedComponent>
</[object Object]>
`;

exports[`Return To Home renders correctly 1`] = `
<TouchableWithoutFeedback>
  <AnimatedComponent
    style={
      Array [
        Object {
          "opacity": 50,
          "transform": Array [
            Object {
              "translateX": 50,
            },
            Object {
              "translateY": 50,
            },
            Object {
              "rotate": 50,
            },
          ],
        },
        Object {
          "alignSelf": "center",
          "padding": 20,
          "position": "absolute",
          "top": 70,
        },
      ]
    }
  >
    <Component
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/icons/leftCarrotWhite.png",
        }
      }
      style={
        Object {
          "height": 28,
          "width": 14,
        }
      }
    />
  </AnimatedComponent>
</TouchableWithoutFeedback>
`;

exports[`Swipe App renders correctly 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <AnimatedComponent
    automaticallyAdjustContentInsets={false}
    bounces={false}
    contentOffset={
      Object {
        "x": 750,
      }
    }
    horizontal={true}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    pagingEnabled={true}
    scrollEventThrottle={16}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Page
      key="page-0"
    >
      <Connect(Account)
        Load={[Function]}
        notificationPermissionInfo={[Function]}
        setNotifsRequested={[Function]}
        switchColor={"rgba(247, 147, 26, 1)"}
        switchXOffset={NaN}
      />
    </Page>
    <Page
      key="page-1"
    >
      <Connect(Home)
        Load={[Function]}
        notificationPermissionInfo={[Function]}
        setNotifsRequested={[Function]}
        switchColor={"rgba(247, 147, 26, 1)"}
        switchXOffset={NaN}
        yOffset={0}
      />
    </Page>
    <Page
      key="page-2"
    >
      <Connect(Wallet)
        Load={[Function]}
        notificationPermissionInfo={[Function]}
        setNotifsRequested={[Function]}
        switchColor={"rgba(247, 147, 26, 1)"}
        switchXOffset={NaN}
      />
    </Page>
  </AnimatedComponent>
  <AnimatedComponent
    style={
      Array [
        Object {
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        },
        Object {
          "overflow": "visible",
          "position": "absolute",
          "shadowOffset": Object {
            "height": 5,
            "width": 0,
          },
          "shadowOpacity": 0.12,
          "shadowRadius": 12,
          "top": -40,
          "width": 750,
        },
      ]
    }
  >
    <Header
      fill={"rgba(247, 147, 26, 1)"}
      fillInput={NaN}
    />
  </AnimatedComponent>
  <TouchableWithoutFeedback
    key="page-icon-0"
    onPressIn={[Function]}
  >
    <AnimatedComponent
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/icons/whiteSplash.png",
        }
      }
      style={
        Array [
          Object {
            "alignSelf": "center",
            "marginBottom": 15,
            "overflow": "visible",
            "paddingLeft": 20,
            "paddingRight": 20,
            "position": "absolute",
            "shadowOffset": Object {
              "height": 5,
              "width": 0,
            },
            "shadowOpacity": 0.12,
            "shadowRadius": 12,
            "top": 55,
          },
          Object {
            "transform": Array [
              Object {
                "scale": 0.75,
              },
              Object {
                "translateX": -412.50000000000006,
              },
            ],
          },
          Object {
            "height": 45,
            "width": 45,
          },
          Object {},
        ]
      }
    />
  </TouchableWithoutFeedback>
  <TouchableWithoutFeedback
    key="page-icon-2"
    onPressIn={[Function]}
  >
    <AnimatedComponent
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/icons/qrIcon.png",
        }
      }
      style={
        Array [
          Object {
            "alignSelf": "center",
            "marginBottom": 15,
            "overflow": "visible",
            "paddingLeft": 20,
            "paddingRight": 20,
            "position": "absolute",
            "shadowOffset": Object {
              "height": 5,
              "width": 0,
            },
            "shadowOpacity": 0.12,
            "shadowRadius": 12,
            "top": 55,
          },
          Object {
            "transform": Array [
              Object {
                "scale": 0.75,
              },
              Object {
                "translateX": 412.50000000000006,
              },
            ],
          },
          Object {
            "height": 37,
            "width": 37,
          },
          Object {
            "top": 59,
          },
        ]
      }
    />
  </TouchableWithoutFeedback>
  <AnimatedComponent
    key="swipe-app-title-0"
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "position": "absolute",
          "top": 115,
          "width": 750,
        },
        Object {
          "transform": Array [
            Object {
              "scale": 0.75,
            },
            Object {
              "translateX": -900,
            },
          ],
        },
      ]
    }
  >
    <Component
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 24,
          "fontWeight": "700",
        }
      }
    >
      @undefined
    </Component>
  </AnimatedComponent>
  <AnimatedComponent
    key="swipe-app-title-2"
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "position": "absolute",
          "top": 115,
          "width": 750,
        },
        Object {
          "transform": Array [
            Object {
              "scale": 0.75,
            },
            Object {
              "translateX": 900,
            },
          ],
        },
      ]
    }
  >
    <Component
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 24,
          "fontWeight": "700",
        }
      }
    >
      Your splash wallet
    </Component>
  </AnimatedComponent>
  <Connect(ReturnToHome)
    onPress={[Function]}
    xOffset={750}
    yOffsets={
      Object {
        "home": 0,
      }
    }
  />
  <Connect(Balance)
    xOffset={750}
    yOffsets={
      Object {
        "home": 0,
      }
    }
  />
  <Connect(CurrencySwitch)
    switchXOffset={NaN}
    xOffset={750}
    yOffsets={
      Object {
        "home": 0,
      }
    }
  />
  <Connect(ModalRoot) />
</Component>
`;
